src/main/java/Problem1/ArrayDictionary.java

```diff
@@ -46,6 +46,16 @@ public void put(String key, String value) {
     @Override
     public void remove(String key) {
         // homework
+        int hashedKey = hashFunction(key);
+        KVEntry ptr = entries[hashedKey];
+        while (ptr.next != null) {
+            if (ptr.key.equals(key)) {
+                ptr.value = null;
```

Comment:
incorrect. 

the code here should be very similar to removing a node from a single linked list.